confidential-data-hub/hub/src/bin/client-tool/main.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2024 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! This tool is to test ttrpc Confidential Data Hub
+
+#![allow(non_snake_case)]
+
+use api::*;
+use api_ttrpc::{GetResourceServiceClient, SealedSecretServiceClient, SecureMountServiceClient};
+use base64::{engine::general_purpose::STANDARD, Engine};
+use clap::{Args, Parser, Subcommand};
+use keyprovider::*;
+use keyprovider_ttrpc::KeyProviderServiceClient;
+use storage::volume_type::Storage;
+use ttrpc::context;
+
+mod api;
+mod api_ttrpc;
+
+mod keyprovider;
+mod keyprovider_ttrpc;
+
+#[derive(Parser)]
+#[command(name = "cdh_client")]
+#[command(bin_name = "cdh_client")]
+#[command(author, version, about, long_about = None)]
+struct Cli {
+    #[command(subcommand)]
+    operation: Operation,
+
+    /// ttrpc socket path
+    #[arg(short, long, default_value_t = String::from("unix:///run/confidential-containers/cdh.sock"))]
+    socket: String,
+}
+
+#[derive(Subcommand)]
+#[command(author, version, about, long_about = None)]
+enum Operation {
+    /// Unseal the given sealed secret
+    UnsealSecret(UnsealSecretArgs),
+
+    /// Unwrap the image encryption key
+    UnwrapKey(UnwrapKeyArgs),
+
+    /// Get Resource from KBS
+    GetResource(GetResourceArgs),
+
+    /// Secure mount
+    SecureMount(SecureMountArgs),
+}
+
+#[derive(Args)]
+#[command(author, version, about, long_about = None)]
+struct UnsealSecretArgs {
+    /// path to the file which contains the sealed secret
+    #[arg(short, long)]
+    secret_path: String,
+}
+
+#[derive(Args)]
+#[command(author, version, about, long_about = None)]
+struct UnwrapKeyArgs {
+    /// path to the file which contains the AnnotationPacket
+    #[arg(short, long)]
+    annotation_path: String,
+}
+
+#[derive(Args)]
+#[command(author, version, about, long_about = None)]
+struct GetResourceArgs {
+    /// KBS Resource URI to the target resource
+    #[arg(short, long)]
+    resource_uri: String,
+}
+
+#[derive(Args)]
+#[command(author, version, about, long_about = None)]
+struct SecureMountArgs {
+    /// path to the file which contains the Storage object.
+    #[arg(short, long)]
+    storage_path: String,
+}
+
+#[tokio::main]
+async fn main() {
+    let args = Cli::parse();
+    let inner = ttrpc::asynchronous::Client::connect(&args.socket).expect("connect ttrpc socket");
+
+    match args.operation {
+        Operation::UnsealSecret(arg) => {
+            let client = SealedSecretServiceClient::new(inner);
+            let secret = tokio::fs::read(arg.secret_path).await.expect("read file");
+            let req = UnsealSecretInput {
+                secret,
+                ..Default::default()
+            };
+            let res = client
+                .unseal_secret(context::with_timeout(50 * 1000 * 1000 * 1000), &req)
```

Comment:
Maybe some option to configure this timeout would be nice. At least maybe we should have it be a constant rather than doing the math in each of these cases.

Reply:
Sounds good. Let me add a parameter